File: array-hash/contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> u;
        for (int i = 0; i < nums.size(); ++i)
        {
            if (u.find(nums[i]) == u.end())
            {
                u.insert(nums[i]);
            }

            else return true;
        }
        return false;
    }
};

File: array-hash/contains-duplicate.py
class Solution(object):
    """
    - story (of problem in terms of what should do): check if has duplicccate
    - input: integer array
    - output: boolean
    - Requirements: return true if there is duplicate, false if all unique
    - Constraints that could help:
    - Algo (thinking of): use hashSet to track what you have seen 
- */
    """
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        numsSeen = set()
        for i in nums:
            if i in numsSeen:
                return True
            numsSeen.add(i)
        return False
